src/socialnetwork/ReportHandler.java

```
@@ -0,0 +1,30 @@
+package socialnetwork;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class ReportHandler
+{
+    private static ReportHandler instance;
+    public static ReportHandler getInstance()
+    {
+        if(instance == null) instance = new ReportHandler();
+        return instance;
+    }
+
+
+    private final Queue<Post> reportQueue = new LinkedList<>();
+
+    public void ReceiveReport(Post post)
+    {
+        System.out.println("Thank you for your report. A moderator will look at it soon. Probably.");
+        if (!reportQueue.contains(post)) reportQueue.add(post);
+    }
+
+    public Post GetReport() //should be accessed from moderator. CAUTION: may return null.
```

- Methods should be camelCase (Oracle - 9 Naming Conventions)
- Might want to throw an exception rather than returning null, example of exception could be:
```java
public class ReportNotFoundException extends RuntimeException {
    public ReportNotFoundException(String message) {
        super(message);
    }
}
```